app/models/concerns/domain/lockable.rb

```diff
@@ -0,0 +1,53 @@
+module Concerns::Domain::Lockable
+  extend ActiveSupport::Concern
+
+  def apply_registry_lock
```

Comment:
Do we have any other lock besides registry? Simply `#lock` does not work?

Reply:
There's an `ActiveRecord` method called `lock!`, I'd rather avoid such confusion.